Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h

```diff
@@ -0,0 +1,286 @@
+  millePedeResFile_(config.getParameter<std::string>("millePedeResFile")),
+
+  sigCut_     (config.getParameter<double>("sigCut")),
+  Xcut_       (config.getParameter<double>("Xcut")),
+  tXcut_      (config.getParameter<double>("tXcut")),
+  Ycut_       (config.getParameter<double>("Ycut")),
+  tYcut_      (config.getParameter<double>("tYcut")),
+  Zcut_       (config.getParameter<double>("Zcut")),
+  tZcut_      (config.getParameter<double>("tZcut")),
+  maxMoveCut_ (config.getParameter<double>("maxMoveCut")),
+  maxErrorCut_(config.getParameter<double>("maxErrorCut"))
+{
+}
+
+void MillePedeFileReader
+::read() {
+  readMillePedeLogFile();
+  readMillePedeResultFile();
+}
+
+bool MillePedeFileReader
+::storeAlignments() {
+  return updateDB;
+}
+
+
+
+//=============================================================================
+//===   PRIVATE METHOD IMPLEMENTATION                                       ===
+//=============================================================================
+
+void MillePedeFileReader
+::readMillePedeLogFile()
+{
+  std::ifstream logFile;
+  logFile.open(millePedeLogFile_.c_str());
+
+  if (logFile.is_open()) {
+    edm::LogInfo("MillePedeFileReader") << "Reading millepede log-file";
+    std::string line;
+
+    while (getline(logFile, line)) {
+      std::string Nrec_string = "NREC =";
+
+      if (line.find(Nrec_string) != std::string::npos) {
+        std::istringstream iss(line);
+        std::string trash;
+        iss >> trash >> trash >> Nrec;
+
+        if (Nrec < 25000) {
+          PedeSuccess = false;
+          Movements   = false;
+          Error       = false;
+          Significant = false;
+          updateDB   = false;
+        }
+      }
+    }
+
+  } else {
+    edm::LogError("MillePedeFileReader") << "Could not read millepede log-file.";
+
+    PedeSuccess = false;
+    Movements   = false;
+    Error       = false;
+    Significant = false;
+    updateDB   = false;
+    Nrec = 0;
+  }
+}
+
+void MillePedeFileReader
+::readMillePedeResultFile()
+{
+  std::ifstream resFile;
+  resFile.open(millePedeResFile_.c_str());
+
+  if (resFile.is_open()) {
+    edm::LogInfo("MillePedeFileReader") << "Reading millepede result-file";
+    double Multiplier[6] = {10000.,10000.,10000.,1000000.,1000000.,1000000.};
+
+    std::string line;
+    getline(resFile, line); // drop first line
+
+    while (getline(resFile, line)) {
+      std::istringstream iss(line);
+
+      std::vector<std::string> tokens;
+      std::string token;
+      while (iss >> token) {
+        tokens.push_back(token);
+      }
+
+      if (tokens.size() > 4 /*3*/) {
+        PedeSuccess = true;
+
+        int alignable      = std::stoi(tokens[0]);
+        int alignableIndex = alignable % 10 - 1;
+
+        double ObsMove = std::stof(tokens[3]) * Multiplier[alignableIndex];
+        double ObsErr  = std::stof(tokens[4]) * Multiplier[alignableIndex];
+
+        int det = -1;
+
+        if (alignable >= 60 && alignable <= 69) {
```

Comment:
I think, as long as the number of alignables does not exceed 700000 we are fine:

https://github.com/cerminar/cmssw/blob/PCLDQMPlots_80X_v2/Alignment/MillePedeAlignmentAlgorithm/src/PedeLabelerBase.cc#L22

Haven't checked it, though.